src/tasks/Common/Utils.cs

```diff
@@ -411,4 +414,33 @@ private static bool IsManagedAssembly(PEReader peReader)
             return false;
         }
     }
+
+    // Keep synced with mono_fixup_symbol_name from src/mono/mono/metadata/native-library.c
+    public static string FixupSymbolName(string name)
+    {
+        UTF8Encoding utf8 = new();
+        byte[] bytes = utf8.GetBytes(name);
+        StringBuilder sb = new();
+
+        foreach (byte b in bytes)
+        {
+            if ((b >= (byte)'0' && b <= (byte)'9') ||
+                (b >= (byte)'a' && b <= (byte)'z') ||
+                (b >= (byte)'A' && b <= (byte)'Z') ||
+                (b == (byte)'_'))
+            {
+                sb.Append((char)b);
+            }
+            else if (s_charsToReplace.Contains((char)b))
+            {
+                sb.Append('_');
+            }
+            else
+            {
+                sb.Append($"_{b:X}_");
```

Comment:
I'm not sure of the exact constraints for this name mapping algorithm but could we just replace all chars outside of a-z/0-9 with underscore? that would allow us to not allocate in the native side

Reply:
This is needed when mapping two assemblies to the same name.
For example having an app project that references a library project named in a way that maps to the same name, but is different before fixup.

Reply:
Ah so e.g. `HelloÖProject.DoSomething()` and `HelloÜProject.DoSomething()` would both map to `Hello_Project_DoSomething`, makes sense.

Reply:
not sure whether it'd be strictly better but we could also collect the special chars and append them at the end or return them separately. that'd have the benefit of not needing to allocate when the string doesn't have special chars

@lambdageek for thoughts

Reply:
I'm not sure how adding the characters in a suffix would make a difference.

On the native side I guess it would be nice to have a function that precomputes the required space for the entire mangled name and just does a single allocation.  (ie: get rid of the `g_strdup_printf` with the prefix and the name together with the mangling loop - and just make a helper that that does a single allocation for the final mangled name)